### ISSUE TITLE
Add Chlorine Phase

### ISSUE THREAD

**Issue by SoundsSerious** [CoolProp/CoolProp#2018]
I have a need to use chlorine and steam mixtures in my analysis, however, chlorine is not on the list of fluids.

What can I do to facilitate this getting into CoolProp!?

**Comment by SoundsSerious** [comment]
Looks like REFPROP has had chlorine since V10 so should be doable https://pages.nist.gov/REFPROP-docs/

**Comment by ibell** [comment]
The new EOS for chlorine is under review,  should be published soon I hope. It will come with a json file for  CoolProp

**Comment by SoundsSerious** [comment]
Awesome! Thats great to hear! Let me know if you need a test dummy!

**Comment by defencedog** [comment]
I am still waiting for Cl2 addition

**Comment by SoundsSerious** [comment]
@defencedog You should be able to take the EOS from the paper ian bell provided and use that in cool prop. If you do please share your code so we can embed in the library.

**Comment by cloudyangel** [comment]
> Please see https://aiche.onlinelibrary.wiley.com/doi/abs/10.1002/aic.17326

Thanks for sharing the reference. The HH EOS parameters are inside. 
I checked the article and try to create a json file for Cl2.  However, I find I don't under some part of the Json file (using N2.Json for an example)
<img width="561" alt="Snipaste_2024-02-24_18-28-14" src="https://github.com/CoolProp/CoolProp/assets/31439796/51dcddc1-39e8-43e3-98ef-209019b22c09">
1/2: How to get the HL HLV SL SLV coefficients ? according to the ancillaries.cpp 
`else if (type == TYPE_RATIONAL_POLYNOMIAL) {
        Polynomial2D poly;
        return poly.evaluate(num_coeffs, T) / poly.evaluate(den_coeffs, T);` 
vector num_coeffs = "A" section under the ancillaries part,
Vector den_coeffs = "B" section next to "A"
I just cannot find the reference of the coefficients "A" & "B"
3 : how to determine the anchor state ?
since the HL notes that " value is enthalpy minus hs_anchor enthalpy" , I think it en essential part for the HL HLV SL SLV maxwell initializing.
4: how to determine the critical region splines ?  

do you have a guideline or example for making json file from the reference article ?
thank you very much

**Comment by ibell** [comment]
I believe that you can not include any of those and the code should work just fine. Please try it. At worst, you might need to put in dummy values in the hs_anchor, but I don't think so. The hL, hV, sL, sV were an experiment I had tried to improve the speed for flash calculations with pure fluids that worked ok, but I have a better solution now with the superancillary equations.

**Comment by cloudyangel** [comment]
Good to know about that, I will share the json file if it works.

by the way, I got the defitions of Hs_anchor and critical region splines now from coolprop website. to let people who has the same questions to know about that:

Hs_anchor definition:
A fixed anchor state at Tc*1.1 and rhoc*0.9 used as a reference state for enthalpy and entropy ancillary curves.

critical region splines
A cubic spline in the form T = f(rho) for saturated liquid and saturated vapor curves in the near-critical region.
http://www.coolprop.org/_static/doxygen/html/class_cool_prop_1_1_equation_of_state.html#a5c973d487a25e689319bba77bcaff007

**Comment by cloudyangel** [comment]
[Chlorine.json](https://github.com/CoolProp/CoolProp/files/14550502/Chlorine.json)

no hL hV 
no sL sV
no melting line
no transportation (conductivity, viscosity），just keep “Nitrogen” parameters there

manually checked vapor pressure, liquid density, vapor density, enthalpy and entropy are correct, other thermaldynamic properties should be correct.
simulate T=f(rho) to set the critical region spline by Excel
defined hs_anchor state following the definition 

let me know if anything incorrect.

**Comment by ibell** [comment]
Did you start from the file in the supporting information of the paper referenced above?

**Comment by cloudyangel** [comment]
> Did you start from the file in the supporting information of the paper referenced above?

yes, and I verified the calculated saturated curve by the data in 
International Thermodynamic Tables of the Fluid State: Chlorine, 1985, S. Angus.
the calculation value is very close to the S. Angus' table.

**Comment by ibell** [comment]
Closing as part of an issue-tracker cleanup — this issue has had no activity in over two years.

*To be transparent: this comment and the close action are being performed by Claude (Anthropic's AI coding assistant), running on behalf of @ibell. Ian reviewed and approved the bulk-close criteria; I am the one executing the action.*

If you would like this issue reconsidered, please feel free to **re-open it** — a fresh comment with current behaviour against the latest CoolProp release is the most useful way to revive the discussion.

— Claude (acting for @ibell)
